migrate to uv

iQuxLE opened the pull request:
###  Migrating to uv

addresses #2615 

- [ ] Update CI/CD
- [ ] Change build system (toml)
- [ ] Update Dockerfile
- [ ] Update Makefiles
- [ ] Generate uv.lock and Remove poetry.lock
- [ ] test migration

dalito reviewed (changes requested):
Thanks for continuing the work on moving to uv.

iQuxLE commented:
I've also fixed two tests 

**Pydantic Array Test**
(`test_arrays_anyshape_json_schema`)

**_Issue_**:  JSON schema self-reference changed in newer Pydantic, causing a test to fail.
**_Fix_**:  Updated the test to be more flexible, checking for "AnyShapeArray" within the self-reference.

**Deprecation Tests**
(`test_markdowngen_deprecation` and `test_yumlgen_deprecation`)

**_Issue_**:  Deprecation warnings weren't showing in development (version 0.0.0).

_**Root Cause:**_:  Version 0.0.0 comes from the fallback-version in pyproject.toml when no git tags exist in this fork, and deprecation logic only triggers when current version ≥ deprecated_in version.

**_Fix_**:  Added skip conditions for these tests when running in a 0.0.0 version environment.

dalito commented:
I don't feel very comfortable to hide the warnings/errors for version 0.0.0.  To "fix" the local deprecation warnings, it should be sufficient to fetch the tags from remote (upstream) 
```bash
git remote add upstream https://github.com/linkml/linkml
git fetch upstream --tags
```
We could add a note to the docs on the importance of tags for calculating the dynamic version.

iQuxLE commented:
I see your point.
Don't you think this overcomplicates things for community devs?

But I would be ok with adding a clear note in the test output if it fails.

dalito reviewed (approved):
Although I would not add it, I don't have a very strong opinion on skipping the version 0.0.0 case. Maybe someone else from the team can also comment.

This looks really nice now! The only thing that you could still do from my point of view is to add a note on fetching the tags to contributing.md. But I'll approve as is.
